lib/dradis/plugins/netsparker/importer.rb

```diff
@@ -34,34 +34,39 @@ def process_report_host(xml_host)
       # Create Nodes from the <url> tags
       host_node_label = xml_host.at_xpath('./url').text
       host_node_label = URI.parse(host_node_label).host rescue host_node_label
-      logger.info{ "\t\t => Creating new host: #{host_node_label}" }
+      logger.info { "\t\t => Creating new host: #{host_node_label}" }
       host_node = content_service.create_node(label: host_node_label, type: :host)
 
       @doc.xpath('/netsparker/vulnerability').each do |xml_vuln|
         process_vuln(xml_vuln, host_node)
       end
-      
     end
 
     def process_vuln(xml_vuln, host_node)
-      type = xml_vuln.at_xpath('./type').text()
+      type = xml_vuln.at_xpath('./type').text
 
       # Create Issues using the Issue template
-      logger.info{ "\t\t => Creating new Issue: #{type}" }
+      logger.info { "\t\t => Creating new Issue: #{type}" }
 
       issue_text = template_service.process_template(template: 'issue', data: xml_vuln)
       issue = content_service.create_issue(text: issue_text, id: type)
 
       # Create Evidence using the Evidence template
       # Associate the Evidence with the Node and Issue
-      logger.info{ "\t\t => Creating new evidence" }
+      logger.info { "\t\t => Creating new evidence" }
       evidence_content = template_service.process_template(
         template: 'evidence', data: xml_vuln
       )
+
+      url = xml_vuln.at_xpath('./url').text
+      node =
+        # If the URL is a valid URI and is not the same as the host_node, create a new node
+        if url =~ URI::ABS_URI && URI(url).host != host_node.label
```

Comment:
The combination of `=` and `if` isn't great, particularly when later you also have to use `||` in L68.

Have you considered either doing a more robust `if` that accounts for the received `host_node` or extracting this to a helper method (e.g. `#get_node` or `#node_for`)?